api/application/src/user/update.rs

```diff
@@ -1,21 +1,33 @@
 use diesel::prelude::*;
 use diesel::Connection;
+use domain::enums::role::Role;
 use domain::models::user::GetUser;
 use domain::models::user::{NewUser, User};
 use infrastructure::ServerState;
-use rocket::response::status::{Created, NotFound};
+use rocket::http::Status;
+use rocket::response::status::Created;
 use rocket::serde::json::Json;
 use rocket::State;
 use shared::response_models::ResponseMessage;
+use shared::token::JWT;
 use uuid::Uuid;
 
 pub fn update_user(
```

Comment:
User is currently able to update their own role from User to Admin as a check for this is missing.